Add usage examples (#219)
Quansight-Labs/free-threaded-compatibility: docs/examples/mandelbrot.md

# Mandelbrot Set Visualization

!!! note

    See [the worked example notebook accompanying this page](mandelbrot-threads.ipynb)

The [Mandelbrot set](https://en.wikipedia.org/wiki/Mandelbrot_set) is a classic
example of a [fractal](https://en.wikipedia.org/wiki/Fractal) - a mathematical
structure characterized by self-similarity and spatial complexity.

One way to compute a visualization of the set requires looping over coordinates
in the complex plane corresponding to the centers of pixels in an image. Whether
or not a point in the complex plane is in the mandelbrot set is a function only
of the point's coordinates, so visualizing the set is very amenable to parallel
speedups by breaking up the work into chunks of pixels.

> Algorithms to visualize the set range in complexity and sophistication. Here is

Reviewer: ```suggestion
Different algorithms for set visualization range in complexity and sophistication. Here is
```
